core/src/main/java/io/questdb/cairo/sql/PageAddressCache.java

```diff
@@ -50,11 +57,21 @@ public void add(int frameIndex, @Transient PageFrame frame) {
             return; // The page frame is already cached
         }
         for (int columnIndex = 0; columnIndex < columnCount; columnIndex++) {
-            pageAddresses.add(frame.getPageAddress(columnIndex));
+            final long pageAddress = frame.getPageAddress(columnIndex);
+            pageAddresses.add(pageAddress);
             int varSizeColumnIndex = varSizeColumnIndexes.getQuick(columnIndex);
             if (varSizeColumnIndex > -1) {
-                auxPageAddresses.add(frame.getIndexPageAddress(columnIndex));
-                pageSizes.add(frame.getPageSize(columnIndex));
+                final long auxPageAddress = frame.getIndexPageAddress(columnIndex);
+                auxPageAddresses.add(auxPageAddress);
+                final long pageSize = frame.getPageSize(columnIndex);
+                pageSizes.add(pageSize);
+                final int columnFrameIndex = columnCount * frameIndex + columnIndex;
+                pageLimits.extendAndSet(columnFrameIndex, pageAddress + pageSize);
+                final long frameRowCount = frame.getPartitionHi() - frame.getPartitionLo();
+                varcharAuxPageLimits.extendAndSet(
```

Comment:
for sanity of this code, we should check the type of column before populating it. Also aux vector size for the row count is the same for every column. There is no need to calculate it for each column in this "for" loop